vocode/turn_based/synthesizer/coqui_synthesizer.py

```diff
@@ -1,29 +1,113 @@
 import io
-from typing import Optional
+from typing import Optional, List
 from pydub import AudioSegment
 import requests
 from vocode import getenv
 from vocode.turn_based.synthesizer.base_synthesizer import BaseSynthesizer
+import aiohttp
+import asyncio
 
 COQUI_BASE_URL = "https://app.coqui.ai/api/v2/"
 DEFAULT_SPEAKER_ID = "d2bd7ccb-1b65-4005-9578-32c4e02d8ddf"
+MAX_TEXT_LENGTH = 250  # The maximum length of text that can be synthesized at once
 
 
 class CoquiSynthesizer(BaseSynthesizer):
-    def __init__(self, voice_id: Optional[str] = None, api_key: Optional[str] = None):
+    def __init__(
+        self,
+        voice_id: Optional[str] = None,
+        voice_prompt: Optional[str] = None,
+        api_key: Optional[str] = None,
+    ):
         self.voice_id = voice_id or DEFAULT_SPEAKER_ID
+        self.voice_prompt = voice_prompt
         self.api_key = getenv("COQUI_API_KEY", api_key)
 
     def synthesize(self, text: str) -> AudioSegment:
-        url = COQUI_BASE_URL + "samples"
-        headers = {"Authorization": f"Bearer {self.api_key}"}
-        body = {
-            "text": text,
-            "speaker_id": self.voice_id,
-            "name": "unnamed",
-        }
+        # Split the text into chunks of less than MAX_TEXT_LENGTH characters
+        text_chunks = self.split_text(text)
+        # Synthesize each chunk and concatenate the results
+        audio_chunks = [self.synthesize_chunk(chunk) for chunk in text_chunks]
+        return sum(audio_chunks) # type: ignore
+
+    def synthesize_chunk(self, text: str) -> AudioSegment:
+        url, headers, body = self.get_request(text)
+
+        # Get the sample
         response = requests.post(url, headers=headers, json=body)
         assert response.ok, response.text
         sample = response.json()
         response = requests.get(sample["audio_url"])
         return AudioSegment.from_wav(io.BytesIO(response.content))  # type: ignore
+
+    def split_text(self, text: str) -> List[str]:
```

Comment:
idea: this implementation may be easier to read? via gpt4

```
def split_text(self, text: str) -> List[str]:
        sentence_enders = re.compile('[.!?]')
        sentences = sentence_enders.split(text)
        chunks = []
        current_chunk = ""
        
        for sentence in sentences:
            sentence = sentence.strip()
            if not sentence:
                continue
            
            proposed_chunk = current_chunk + sentence
            if len(proposed_chunk) > 250:
                chunks.append(current_chunk.strip())
                current_chunk = sentence + "."
            else:
                current_chunk = proposed_chunk + "."
                
        if current_chunk:
            chunks.append(current_chunk.strip())
            
        return chunks
```

Reply:
done